closed/src/java.base/share/native/libjncrypto/NativeCrypto.c

```diff
@@ -1626,6 +1714,59 @@ JNIEXPORT jint JNICALL Java_jdk_crypto_jniprovider_NativeCrypto_RSAEP
     return (jint)msg_len;
 }
 
+/* RSADP Cryptographic Primitive, RSA Private Key operation
+ * Returns -1 on error
+ * The param verify is -1 for 'no verify', otherwise it is size of m (with verify)
+ *
+ * Class:     jdk_crypto_jniprovider_NativeCrypto
+ * Method:    RSADP
+ * Signature: ([BI[BIJ)I
+ */
+
+typedef struct RSADP_BN {
+    BN_CTX *ctx;
+    BIGNUM *k_bigNum;
+    BIGNUM *m_bigNum;
+    BIGNUM *rem;
+} RSADP_BN;
+
+/*
+ * Class:     jdk_crypto_jniprovider_NativeCrypto
+ * Method:    create_rsadp_BN
+ * Signature: ()J
+ */
+JNIEXPORT jlong JNICALL Java_jdk_crypto_jniprovider_NativeCrypto_create_1rsadp_1BN
+	(JNIEnv *env, jclass thisObj) {
+
+    RSADP_BN *rsadp_bn_ptr = (RSADP_BN *)malloc(sizeof(RSADP_BN));
+    if (rsadp_bn_ptr != NULL) {
+        rsadp_bn_ptr->ctx = (*OSSL_BN_CTX_new)();
+        rsadp_bn_ptr->k_bigNum = (*OSSL_BN_new)();
+        rsadp_bn_ptr->m_bigNum = (*OSSL_BN_new)();
+        rsadp_bn_ptr->rem = (*OSSL_BN_new)();
+        return (jlong)rsadp_bn_ptr;
```

Comment:
Casts between `jlong` and pointers really ought to go through `intptr_t` to avoid changing signed-ness and size in one step.
* `jlong` to `T *` : `(T *)(intptr_t)longValue`
* `T *` to `jlong` : `(jlong)(intptr_t)somePointer`